src/main/fulcro/client/alpha/dom.cljc

```diff
@@ -0,0 +1,808 @@
+  radialGradient
+  stop
+  tspan)
+
+(def tags
+  '[a
+    abbr
+    address
+    area
+    article
+    aside
+    audio
+    b
+    base
+    bdi
+    bdo
+    big
+    blockquote
+    body
+    br
+    button
+    canvas
+    caption
+    cite
+    code
+    col
+    colgroup
+    data
+    datalist
+    dd
+    del
+    details
+    dfn
+    dialog
+    div
+    dl
+    dt
+    em
+    embed
+    fieldset
+    figcaption
+    figure
+    footer
+    form
+    h1
+    h2
+    h3
+    h4
+    h5
+    h6
+    head
+    header
+    hr
+    html
+    i
+    iframe
+    img
+    ins
+    kbd
+    keygen
+    label
+    legend
+    li
+    link
+    main
+    map
+    mark
+    menu
+    menuitem
+    meta
+    meter
+    nav
+    noscript
+    object
+    ol
+    optgroup
+    output
+    p
+    param
+    picture
+    pre
+    progress
+    q
+    rp
+    rt
+    ruby
+    s
+    samp
+    script
+    section
+    small
+    source
+    span
+    strong
+    style
+    sub
+    summary
+    sup
+    table
+    tbody
+    td
+    tfoot
+    th
+    thead
+    time
+    title
+    tr
+    track
+    u
+    ul
+    var
+    video
+    wbr
+
+    ;; svg
+    circle
+    clipPath
+    ellipse
+    g
+    line
+    mask
+    path
+    pattern
+    polyline
+    rect
+    svg
+    text
+    defs
+    linearGradient
+    polygon
+    radialGradient
+    stop
+    tspan])
+
+;; (defmacro ^:private gen-react-dom-inline-fns []
+;;   (when (boolean (:ns &env))
+;;     `(do
+;;        ;; ~@(clojure.core/map gen-react-dom-inline-fn tags)
+;;        ~@(clojure.core/map gen-dom-macro tags)
+;;        )))
+
+;; #?(:clj (gen-react-dom-inline-fns))
+
+;; (defn ^:private gen-react-dom-fn [tag]
+;;   `(defn ~tag [opts# & children#]
+;;      {:style/indent 1}
+;;      (.apply ~(symbol "js" "React.createElement") nil
+;;        (cljs.core/into-array
+;;          (cons ~(name tag) (cons opts# (cljs.core/map fulcro.util/force-children children#)))))))
+
+;; (defmacro ^:private gen-react-dom-fns []
+;;   (let [raw-inputs?  (boolean (System/getProperty "rawInputs" nil))
+;;         tags         (if raw-inputs?
+;;                        (concat tags '[input textarea select option])
+;;                        tags)
+;;         extra-inputs (when-not raw-inputs?
+;;                        '[(def input (wrap-form-element "input"))
+;;                          (def textarea (wrap-form-element "textarea"))
+;;                          (def option (wrap-form-element "option"))
+;;                          (def select (wrap-form-element "select"))])]
+;;     `(do
+;;        ~@(clojure.core/map gen-react-dom-fn tags)
+;;        ~@extra-inputs)))
+
+;; ===================================================================
+;; Server-side rendering
+
+;; https://github.com/facebook/react/blob/57ae3b/src/renderers/dom/shared/SVGDOMPropertyConfig.js
+;; https://github.com/facebook/react/blob/57ae3b/src/renderers/dom/shared/HTMLDOMPropertyConfig.js
+#?(:clj
+   (def supported-attrs
```

Comment:
React itself dropped attribute validations (https://reactjs.org/blog/2017/09/26/react-v16.0.html#support-for-custom-dom-attributes) because they prevent you from adopting new features (like hurting the open principle). So I think we are better not validating it, what you guys think?

Reply:
I agree